src/pages/ProductItemPage/ProductItemPage.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useState } from 'react';
+import { getPhones } from '../../api';
+import { useParams } from 'react-router-dom';
+import { ProductItemType } from '../../types/ProductItemType';
+import { NotFoundPage } from '../NotFoundPage';
+import { ButtonColor } from '../../components/UI/ButtonColor';
+import styles from './productItemPage.module.scss';
+
+export const ProductItemPage = () => {
+  const { productId } = useParams<{ productId: string }>();
+  const [product, setProduct] = useState<ProductItemType | null>(null);
+  const [selectedColor, setSelectedColor] = useState(
+    product?.colorsAvailable[0],
+  );
+  const [selectedCapacity, setSelectedCapacity] = useState(
+    product?.capacityAvailable[0],
+  );
+
+  useEffect(() => {
+    const fetchProduct = async () => {
+      try {
+        const phones = await getPhones();
+        const foundProduct = phones.find(phone => phone.id === productId);
+        if (foundProduct) {
+          setProduct(foundProduct);
+        } else {
+          setProduct(null);
+        }
+      } catch (error) {
+        console.error('Error fetching product');
+      }
+    };
+
+    if (productId) {
+      fetchProduct();
+    }
+  }, [productId]);
+
+  const handleColorChange = (color: string) => {
+    setSelectedColor(color);
+  };
+
+  const handleCapacityChange = (capacity: string) => {
+    setSelectedCapacity(capacity);
+  };
+
+  return (
+    <div className="product-item">
+      {product ? (
+        <>
+          <h1 className="title">{product.name}</h1>
+          <div className="product-details">
+            <div className="product-images">
+              {product.images.map((image, index) => (
+                <img
+                  key={index}
+                  src={image}
+                  alt={product.name}
```

Comment:
Need to change this part, it's okey for skelet, but in the future^
key = should have not index, need something another
alt = need to change too